--- a/build/target/product/core.mk	2012-08-10 03:24:15.964912682 +0000
+++ b/build/target/product/core.mk	2012-08-10 03:41:07.432911481 +0000
@@ -22,6 +22,7 @@ PRODUCT_PROPERTY_OVERRIDES := \
     ro.config.notification_sound=OnTheHunt.ogg \
     ro.config.alarm_alert=Alarm_Classic.ogg
 
+# BEGIN privacy-added: PDroid
 PRODUCT_PACKAGES := \
     ApplicationsProvider \
     BackupRestoreConfirmation \
@@ -35,6 +36,7 @@ PRODUCT_PACKAGES := \
     Home \
     KeyChain \
     MediaProvider \
+    PDroid \
     PackageInstaller \
     PicoTts \
     SettingsProvider \
@@ -120,6 +123,7 @@ PRODUCT_PACKAGES := \
     sensorservice \
     lint \
     uiautomator
+# END privacy-added: PDroid
 
 PRODUCT_COPY_FILES += \
     system/core/rootdir/init.usb.rc:root/init.usb.rc \
