--- a/frameworks/base/test-runner/Android.mk	2012-08-10 03:25:00.216912681 +0000
+++ b/frameworks/base/test-runner/Android.mk	2012-08-10 04:15:32.820912682 +0000
@@ -20,7 +20,7 @@ include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(call all-java-files-under, src)
 
-LOCAL_JAVA_LIBRARIES := core core-junit framework
+LOCAL_JAVA_LIBRARIES := core core-junit framework framework2
 
 LOCAL_MODULE:= android.test.runner
 
