--- cyanogen/frameworks/base/services/java/com/android/server/am/ActivityManagerService.java	2012-07-13 09:01:37.408330957 -0700
+++ pang/frameworks/base/services/java/com/android/server/am/ActivityManagerService.java	2012-07-14 21:21:24.375401354 -0700
@@ -12771,6 +12771,7 @@ public final class ActivityManagerServic
                                 r.resultExtras, r.resultAbort, true);
                         if (doNext) {
                             doTrim = true;
+                            r.queue.setContext(mContext);
                             r.queue.processNextBroadcast(false);
                         }
                     }
@@ -13278,6 +13279,7 @@ public final class ActivityManagerServic
             }
 
             if (doNext) {
+                r.queue.setContext(mContext);
                 r.queue.processNextBroadcast(false);
             }
             trimApplications();
